astropy/io/registry.py

```diff
@@ -137,7 +137,7 @@ def get_formats(data_class=None, readwrite=None):
     format_table = Table(data, names=('Data class', 'Format', 'Read', 'Write',
                                       'Auto-identify', 'Deprecated'))
 
-    if not np.any(format_table['Deprecated'] == 'Yes'):
+    if not np.any('Yes' in format_table['Deprecated']):
```

Comment:
shouldn't this also raise the warning when the dtypes differ:
```
In [28]: b = Table([[1,2,3],['a','b','c']])

In [29]: b['col0'] == 'a'
/Users/bsipocz/munka/devel/astropy/astropy/table/column.py:975: FutureWarning: elementwise comparison failed; returning scalar instead, but in the future will perform elementwise comparison
  return getattr(self.data, op)(other)
Out[29]: False

In [30]: 'a' in b['col0']
/Users/bsipocz/munka/devel/astropy/astropy/table/column.py:975: FutureWarning: elementwise comparison failed; returning scalar instead, but in the future will perform elementwise comparison
  return getattr(self.data, op)(other)
Out[30]: False

In [31]: b['col1'] == 'a'
Out[31]: array([ True, False, False])
```

Reply:
Good point. Maybe this is not the correct solution then.